### DISCUSSION TITLE
When using the integration with //wsl$ paths is it possible to use Windows native edit/merge/diff tools

### DISCUSSION THREAD

**Discussion by dude0001** [gitextensions/gitextensions#10498] [Q&A]
I am using WSL and have Git Extension opening a repo cloned to a path in `//wsl$` This causes all `git` commands to run as `wsl git` inside my Linux distro running in WSL. This is a great feature as it improves performance, thank you! One pain point I do have with this is I still want to use native Windows tools for doing edit/merge/diff commands. What happens instead for example when Git needs an editor vi is launched inside an embedded terminal within Git Extensions. This might be desirable for some, but I'd like instead VS Code to be launched from my Windows host when Git prompts for an editor. Is this possible or should I enter an issue for the feature request?

Edit:

I did find I can change the editor in my .gitconfig to use the `code` command instead of specifying the full Windows path. This causes VS Code to launch on my Windows host in remote WSL. I'm still not sure how to do with other tools I want to use for diff/merge.

```
[core]
	editor = code --wait
```

**Comment by gerhardol** [comment]
In WSL, the Windows diff/merge tools is only used for the custom list with all tools. The native wsl Git --gui tool is used by default.
If you want the Win tools without using the custom list, set it up in WSL Git:

```
[mergetool "tortoisemerge"]
        path = /mnt/c/Program Files/TortoiseGit/bin/TortoiseGitMerge.exe
        cmd = \"/mnt/c/Program Files/TortoiseGit/bin/TortoiseGitMerge.exe\" -base:\"$BASE\" -mine:\"$LOCAL\" -theirs:\"$REMOTE\"
```

Using the Win tools is kind of the intention with WSL compared to solutions like VMWare.

**Reply by mstv** [reply]
Viewing diffs using the external `diff.guitool` needs an additional path conversion because temporarily checked out file revisions are stored in the `/tmp/` folder of WSL2 by its git.
This conversion can easily be done using `wslpath -aw`:
```
[difftool "favorite_diff"]
        path = /mnt/c/Program Files/FavoriteDiff/favorite_diff.exe
        cmd = \"/mnt/c/Program Files/FavoriteDiff/favorite_diff.exe\" \"$(wslpath -aw $LOCAL)\" \"$(wslpath -aw $REMOTE)\"
```

In my setup, there is no need to adapt the `mergetool` because the Resolve Conflicts dialog directly calls the merge tool executable.

(Opening a repo in a WSL2 fs using a mapped network-drive letter does not help / work because the WSL2 git always converts them back to WSL paths; and git recommends the wrong path to be added as trusted folder.)

**Reply by gerhardol** [reply]
The primary difftool is the WSL definition and must be usable also from the command line, so you must set /mnt paths too.
The wsl environment is used as some tools may not work with WSL parths, so it is possible to use separate tools in Win and WSL.
I guess this could be configurable.

The list with all defined difftools is Win only though, the expectation is that the wsl list is not configured.
This is not fully consistent. Since WSLg supports GUI OK, this should maybe be changed.
